src/analysis/inner_product_space/pi_L2.lean

```diff
@@ -167,37 +273,58 @@ begin
   { rw [← v.equiv_fun.symm_apply_apply y, v.equiv_fun_symm_apply] }
 end
 
-@[simp] lemma basis.coe_isometry_euclidean_of_orthonormal
-  (v : basis ι 𝕜 E) (hv : orthonormal 𝕜 v) :
-  (v.isometry_euclidean_of_orthonormal hv : E → euclidean_space 𝕜 ι) = v.equiv_fun :=
+@[simp] lemma _root_.basis.coe_to_orthonormal_basis_repr (v : basis ι 𝕜 E) (hv : orthonormal 𝕜 v) :
+  ((v.to_orthonormal_basis hv).repr : E → euclidean_space 𝕜 ι) = v.equiv_fun :=
 rfl
 
-@[simp] lemma basis.coe_isometry_euclidean_of_orthonormal_symm
+@[simp] lemma _root_.basis.coe_to_orthonormal_basis_repr_symm
   (v : basis ι 𝕜 E) (hv : orthonormal 𝕜 v) :
-  ((v.isometry_euclidean_of_orthonormal hv).symm : euclidean_space 𝕜 ι → E) = v.equiv_fun.symm :=
+  ((v.to_orthonormal_basis hv).repr.symm : euclidean_space 𝕜 ι → E) = v.equiv_fun.symm :=
 rfl
 
+@[simp] lemma _root_.basis.coe_to_orthonormal_basis (v : basis ι 𝕜 E) (hv : orthonormal 𝕜 v) :
+  ⇑(v.to_orthonormal_basis hv) = v :=
```

Comment:
Introducing the `orthonormal_basis.to_basis` construction as above should also shorten this lemma.  Add a simp-lemma (currently missing) to `linear_algebra.basis`:
```lean
@[simp] lemma basis.of_equiv_fun_equiv_fun basis.of_equiv_fun v.equiv_fun = v :=
```
and then this fact can be proved quite quickly via another useful fact:
```lean
@[simp] lemma _root_.basis.to_basis_to_orthonormal_basis (v : basis ι 𝕜 E) (hv : orthonormal 𝕜 v) :
  (v.to_orthonormal_basis hv).to_basis = v :=
by simp [basis.to_orthonormal_basis, orthonormal_basis.to_basis]

@[simp] lemma _root_.basis.coe_to_orthonormal_basis (v : basis ι 𝕜 E) (hv : orthonormal 𝕜 v) :
  (v.to_orthonormal_basis hv : ι → E) = (v : ι → E) :=
calc (v.to_orthonormal_basis hv : ι → E)
    = ((v.to_orthonormal_basis hv).to_basis : ι → E) : by rw orthonormal_basis.coe_to_basis
... = (v : ι → E) : by simp
```